src/Common/src/System/Net/Logging/Logging.cs

```diff
@@ -0,0 +1,724 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Security;
+using System.Threading;
+
+namespace System.Net
+{
+    internal class Logging
+    {
+        private static volatile bool s_LoggingEnabled = true;
```

Comment:
@CIPop, when you say it'll always be enabled, do you mean there will always be a listener and tracing will always output to it, or do you mean that the tracing will always be done in case there is a listener?  I'm assuming the latter, in which case I just want to make sure we follow-through on my later comment and use EventSourceWhatever.Log.IsEnabled around call sites to tracing where actual work is being done, e.g. where string concatenation is being done to formulate what's being traced.  Also, when the tracing mechanism is further designed, it'd be great if there was less of a focus on strings being output and more of a focus on integer values or enums or the like, things that are cheaper to trace out, such that enabling tracing doesn't cause a huge hit to perf.